src/main/groovy/grails/plugin/sentry/SpringSecuritySentryUserProvider.groovy

```diff
@@ -37,57 +33,45 @@ class SpringSecurityUserEventProcessor implements EventProcessor {
                                                     'WL-Proxy-Client-IP',
                                                     'rlnclientipaddr']
 
-    SentryConfig config
+    private SentryConfig config
 
-    SpringSecurityUserEventProcessor(SentryConfig config) {
+    SpringSecuritySentryUserProvider(SentryConfig config) {
         this.config = config
     }
 
     def springSecurityService
 
     @CompileStatic(TypeCheckingMode.SKIP)
     @Override
-    SentryEvent process(SentryEvent event, Object hint) {
+    User provideUser() {
         def isLoggedIn = springSecurityService?.isLoggedIn()
 
         if (isLoggedIn) {
             def principal = springSecurityService.getPrincipal()
 
             if (principal != null && principal != 'anonymousUser') {
-                String idPropertyName = 'id'
-                String emailPropertyName = null
-                String usernamePropertyName = 'username'
-                List data = null
-
-                if (config?.springSecurityUserProperties) {
-                    if (config.springSecurityUserProperties.id) {
-                        idPropertyName = config.springSecurityUserProperties.id
-                    }
-                    if (config.springSecurityUserProperties.email) {
-                        emailPropertyName = config.springSecurityUserProperties.email
-                    }
-                    if (config.springSecurityUserProperties.username) {
-                        usernamePropertyName = config.springSecurityUserProperties.username
-                    }
-                    if (config.springSecurityUserProperties.data) {
-                        data = config.springSecurityUserProperties.data
-                    }
-                }
+                String idPropertyName = config.springSecurityUserProperties?.id ?: 'id'
+                String emailPropertyName = config.springSecurityUserProperties?.email
+                String usernamePropertyName = config.springSecurityUserProperties?.username ?: 'username'
+                List<String> data = config.springSecurityUserProperties?.data ?: []
```

Comment:
I ❤️ this code reduction 😃

Reply:
![image](https://user-images.githubusercontent.com/994676/177513496-d4bcdd3d-de8b-4922-b923-f0af7da8aba5.png)